src/test/java/org/mule/tools/apikit/MainAppScaffolderWithExistingConfigApiSyncTest.java

```diff
@@ -65,72 +76,76 @@ public void reScaffold() throws Exception {
-    Diff secondScaffoldingDiff = XMLUnit.compareXML(secondScaffoldingExpected, secondScaffoldingGenerated);
-    assertTrue(secondScaffoldingDiff.identical());
+    ScaffoldingResult rescaffoldResult = scaffoldApiSync(raml, ramlFolderV2, ROOT_RAML_RESOURCE_URL_V2, muleConfigs);
+    verifySuccessfulScaffolding(rescaffoldResult, "rescaffolding-apisync-version/v2/api.xml");
   }
 
-  private ScaffoldingResult scaffoldApiSync(String raml, String ramlFolder, String rootRamlResourceUrl,
-                                            List<MuleConfig> muleConfigs) {
-    ScaffolderContext context = ScaffolderContextBuilder.builder().withRuntimeEdition(RuntimeEdition.CE).build();
-    MainAppScaffolder mainAppScaffolder = new MainAppScaffolder(context);
+  /**
+   *
+   * First it scaffolds an API normally with configuration of APIKit inside the main xml file with version of raml 0.0.1.
+   * Then it simulates the manual changing of file from main xml --> globals xml.
+   * Finally it re-scaffolds with version 0.0.2.
+   * Expected behaviour is that the new scaffolded API has version 0.0.2 in globals.xml and for the main xml it doesn't
+   * add or remove anything.
+   *
+   */
+  @Test
+  public void reScaffoldDifferentVersionsWithGlobal() throws Exception {
+    String raml = "api.raml";
+    String ramlFolderV1 = "src/test/resources/rescaffolding-apisync-version/v1";
+    String ramlFolderV2 = "src/test/resources/rescaffolding-apisync-version/v2";
+    String firstScaffoldingPath = "rescaffolding-apisync-version/v1/api.xml";
+    String secondScaffoldingApiPathV1 = "scaffolder-from-two-apis/simple/src/main/resources/api/api2-with-global/v1/api.xml";
+    String secondScaffoldingGlobalPathV1 =
+        "scaffolder-from-two-apis/simple/src/main/resources/api/api2-with-global/v1/global.xml";
+    String secondScaffoldingApiPathV2 = "scaffolder-from-two-apis/simple/src/main/resources/api/api2-with-global/v2/api.xml";
+    String secondScaffoldingGlobalPathV2 =
+        "scaffolder-from-two-apis/simple/src/main/resources/api/api2-with-global/v2/global.xml";
 
-    ResourceLoader testScaffolderResourceLoader = new TestScaffolderResourceLoader(ramlFolder);
-    ParseResult parseResult =
-        new ParserService().parse(ApiReference.create(rootRamlResourceUrl + raml, testScaffolderResourceLoader));
-    assertTrue(parseResult.success());
+    ScaffoldingResult result = scaffoldApiSync(raml, ramlFolderV1, ROOT_RAML_RESOURCE_URL, null);
+    verifySuccessfulScaffolding(result, firstScaffoldingPath);
+    List<MuleConfig> muleConfigs = changeGlobalsToSeparateFile(secondScaffoldingApiPathV1, secondScaffoldingGlobalPathV1);
+    ScaffoldingResult secondScaffoldingResult = scaffoldApiSync(raml, ramlFolderV2, ROOT_RAML_RESOURCE_URL_V2, muleConfigs);
+    verifySuccessfulScaffolding(secondScaffoldingResult,
+                                new String[] {secondScaffoldingApiPathV2, secondScaffoldingGlobalPathV2});
```

Comment:
No need to pass them as an array, that's the magic of variable number of arguments in `String...`